### PR TITLE
Issue #13345: Enable examples tests for SimplifyBooleanReturnCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for  SimplifyBooleanReturnCheck

**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the ** SimplifyBooleanReturnCheck** in the project. The implementation focuses on ensuring that the SimplifyBooleanReturn checks are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Review by mahfouz72 — CHANGES_REQUESTED** [review]
In summer we added support for `yield` in this check but we didn't update the examples.
Please add code with `yield` 

**Example**: 

```Java
boolean test(int x, boolean c) {
        return switch (x) {
            case 1 -> {
                if (c) { // violation
                    yield true;
                } else {
                    yield false;
                }
            }
            case 2 -> {yield c;}
            default -> false;
        };
    }
```

@romani Please confirm if it is good to do this update in this PR or we should do in a separate one

**Comment by romani** [comment]
@mahfouz72 , what is simplified by Check version of your code?

and we need compact examples of "bad" code, to avoid example snippet to be long.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.simplifybooleanreturn;
+
+// xdoc section -- start
+class Example1 {
+
+  boolean cond;
+  int a;
+  int b;
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java:15]: `int a, b;`

to make it more compact, as it is not a target of user attention.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    if (cond) { // violation, 'Conditional logic can be removed
+      return true;
+    }
+    else {
+      return false;
+    }
+  }
+
+  // Ok, simplified version of check1()
+  boolean check2() {
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java:27]: lets remove comment and rename method
`boolean check1Simplified() {`

please do same for "int choose2() {"

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  boolean cond;
+  int a;
+  int b;
+
+  boolean check1() {
+    if (cond) { // violation, 'Conditional logic can be removed
+      return true;
+    }
+    else {
```
Comment (romani) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java:21]: please format like 
`} else {`

to make it more compact.
same for others

**Review by romani — CHANGES_REQUESTED** [review]
compacting:

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java (hunk 1 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java (hunk 2 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java]: Done

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java (hunk 3 above)
Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/simplifybooleanreturn/Example1.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by romani** [comment]
@mahfouz72 , switch/yes is not compact structure, so extension to put it will make already long example.

I think we can skip yeld example.
Examples not targeting to show all possibilities, they try to show few cases in case wording of description is not very clear.

**Comment by romani** [comment]
We can always followup in next PR if more people would like to have it.

**Pull request merged**